core/tauri-runtime-wry/src/lib.rs

```diff
@@ -3247,13 +3247,27 @@ fn create_webview<T: UserEvent>(
   if window_builder.center {
     let _ = center_window(&window, window.inner_size());
   }
-  let mut webview_builder = WebViewBuilder::new(window)
-    .map_err(|e| Error::CreateWebview(Box::new(e)))?
+
+  let mut webview_builder =
+    WebViewBuilder::new(window).map_err(|e| Error::CreateWebview(Box::new(e)))?;
+
+  // use with_html method if html content can be extracted from url.
+  // else defaults to with_url method
+  webview_builder = if let Some(html_string) = tauri_utils::html::extract_html_content(&url) {
+    webview_builder
+      .with_html(html_string)
+      .map_err(|e| Error::CreateWebview(Box::new(e)))?
+  } else {
+    webview_builder
+      .with_url(&url)
+      .map_err(|e| Error::CreateWebview(Box::new(e)))?
+  };
```

Comment:
`tauri-runtime-wry` shouldn't do these checks at all, instead, `PendingWindow` should have a field `html: Option<String>` and if it is `Some`, we use `with_html`